src/coreComponents/mesh/FaceElementSubRegion.cpp

```diff
@@ -200,6 +219,141 @@ localIndex FaceElementSubRegion::packUpDownMaps( buffer_unit_type * & buffer,
+
+    sizeOfPackedChars += Pack3< DO_PACKING >( buffer,
+                                             var[li],
+                                             unmappedGI.data(),
+                                             var.sizeOfArray( li ) );
+  }
+
+  return sizeOfPackedChars;
+}
+
+// copied from 977
+inline
+localIndex
+Unpack3( buffer_unit_type const *& buffer,
+         ArrayOfArrays< array1d< globalIndex > > & var,
+         localIndex const subArrayIndex,
+         array1d< globalIndex > & unmappedGlobalIndices )
+{
+  localIndex length;
+  localIndex sizeOfUnpackedChars = bufferOps::Unpack( buffer, length );
+
+  var.resizeArray( subArrayIndex, length );
+  unmappedGlobalIndices.resize( length );
+  unmappedGlobalIndices.setValues< serialPolicy >( unmappedLocalIndexValue );
+
+  bool unpackedGlobalFlag = false;
+  for( localIndex a = 0; a < length; ++a )
+  {
+    array1d< globalIndex > & tmp = var( subArrayIndex, a );
+    sizeOfUnpackedChars += bufferOps::Unpack( buffer, tmp );
+  }
+
+  return sizeOfUnpackedChars;
+}
+
+
+// copied from BufferOps_inline 1304
+inline
+localIndex
+Unpack3( buffer_unit_type const *& buffer,
+         ArrayOfArrays< array1d< globalIndex > > & var,
+         array1d< localIndex > & indices,
+         map< localIndex, array1d< globalIndex > > & unmappedGlobalIndices,
+         unordered_map< globalIndex, localIndex > const & globalToLocalMap )
+{
+  localIndex numIndicesUnpacked;
+  localIndex const sizeOfIndicesPassedIn = indices.size();
+
+  localIndex sizeOfUnpackedChars = bufferOps::Unpack( buffer, numIndicesUnpacked );
+
+  GEOS_ERROR_IF( sizeOfIndicesPassedIn != 0 && numIndicesUnpacked != indices.size(),
+                 "number of unpacked indices(" << numIndicesUnpacked << ") does not equal size of "
+                                                                        "indices passed into Unpack function(" << sizeOfIndicesPassedIn );
+
+  indices.resize( numIndicesUnpacked );
+  array1d< globalIndex > unmappedIndices;
+
+  for( localIndex a=0; a<indices.size(); ++a )
+  {
+    globalIndex gi;
+    sizeOfUnpackedChars += bufferOps::Unpack( buffer, gi );
+
+    localIndex & li = indices[a];
+    if( sizeOfIndicesPassedIn > 0 )
+    {
+      GEOS_ERROR_IF( li!=globalToLocalMap.at( gi ),
+                     "global index "<<gi<<" unpacked from buffer does not equal the lookup "
+                                    <<li<<" for localIndex "<<li<<" on this rank" );
+    }
+    else
+    {
+//      GEOS_ERROR("We should not be there");
+      li = globalToLocalMap.at( gi );
+    }
+
+    unmappedIndices.resize( 0 );
+    sizeOfUnpackedChars += Unpack3( buffer,
+                                   var,
+                                   li,
+                                   unmappedIndices );
+
+    if( unmappedIndices.size() > 0 )
+    {
+      unmappedGlobalIndices[li] = unmappedIndices;
+    }
+  }
+  return sizeOfUnpackedChars;
+}
```

Comment:
These need to be moved to `bufferOps`

Reply:
I've also simplified the functions a bit, you can have a look.